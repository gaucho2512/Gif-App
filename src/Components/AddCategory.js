import React, { useState } from 'react'

//comunicacion entre componentes - setCategories viene desde GifExpertApp.js (como un props)
const AddCategory = ( {setCategories} ) => {

 
const [inputValue, setInputValue] = useState('')

const handleInput = (e) => {
     
    setInputValue( e.target.value)
     
}


const handleSubmit = (e) => {
    e.preventDefault()

    //validacion
    if( inputValue.trim().length > 2 ) {
        setCategories( categ => [  inputValue, ...categ]);
        setInputValue('')
    }

    
     
}


    return (

         <form onSubmit={handleSubmit}> 
           <input
              className="form-control"
              type="text"
              placeholder="Buscar GIFS"
              value={inputValue}
              onChange={ handleInput }
            />
         </form>
   
    )
}

export default AddCategory
