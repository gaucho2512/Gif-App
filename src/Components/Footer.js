import React from 'react'

const Footer = () => {
    return (
        <div>
            <footer>
                <div className="contenedor-footer">
                   <div>
                      <p>Pagina Dise√±ada por <a href="https://www.linkedin.com/in/juan-carlos-ariztegui/" target="_blank" >Juan Ariztegui</a></p>
                   </div>
                </div>
            </footer>
        </div>
    )
}

export default Footer
