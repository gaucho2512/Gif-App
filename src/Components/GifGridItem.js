import React from 'react'
import PropTypes from 'prop-types';

export const GifGridItem = ( { title , url} ) => {


  //console.log({id , title , url});

    return (
        <div className="card animate__animated animate__rubberBand">
           <img src={url} alt={title} />
           <p>{title}</p>
        </div>
    )
}

GifGridItem.prototype = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}

export default GifGridItem
