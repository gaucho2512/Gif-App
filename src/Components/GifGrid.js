import React ,{ Fragment} from 'react'
import useFetchGifs from '../hooks/useFetchGifs'
import GifGridItem from './GifGridItem';


const GifGrid = ({category}) => {

    const {data:images ,loading} = useFetchGifs(category)



    return (
        <Fragment>
            <hr/>
             <h3 className="category">{category}</h3>

           {loading && <p className="animate__animated animate__flash">Loading</p>}
            
         <div className="card-grid"> 
           
            {
                   images.map( img => 
                       <GifGridItem 
                       {...img }
                      key={img}

                       />  
                  )
               }
           
        </div>
 
        </Fragment>
    )
}

export default GifGrid
