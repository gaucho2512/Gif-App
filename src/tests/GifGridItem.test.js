import React from 'react';
import { shallow } from 'enzyme';
import {GifGridItem} from '../Components/GifGridItem'


describe('pruebas en <GifGridItem />' , () => {

    const title = 'un titulo'
    const url = 'http://localhost/algo.jpg'

    test('debe de mostrar el componente correctamente' , () => {

        const wrapper = shallow( <GifGridItem title={title} url={url}/>)

        expect( wrapper).toMatchSnapshot();

      })   


})