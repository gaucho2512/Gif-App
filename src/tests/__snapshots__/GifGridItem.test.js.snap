// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifGridItem /> debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__rubberBand"
>
  <img
    alt="un titulo"
    src="http://localhost/algo.jpg"
  />
  <p>
    un titulo
  </p>
</div>
`;
